### ISSUE TITLE
Incorrect implementation of status codes (Rack interface)

### ISSUE THREAD

**Issue by tayler1** [h2o/h2o#1904]
h2o incorrectly produces http response codes with mruby handlers.

```
config
/:
  mruby.handler: lambda {[500,[]]}

server response
-> "HTTP/1.1 500 OK\r\n"
```

**Comment by utrenkner** [comment]
What would you have expected? With Rack you should generally return a status code, the header, and the body, e.g. 
```
[200, {'content-type' => 'text/plain'}, ["Hello world\n"]]
```
If you return something with status code 500, the server will use that as status code of its response. But maybe I did not fully understand your question?

**Comment by fetus-hina** [comment]
Maybe she/he wants:
```diff
- HTTP/1.1 500 OK
+ HTTP/1.1 500 Internal Server Error
```

**Comment by kazuho** [comment]
The field has no meaning. Quoting from [RFC 7230 section 3.1.2](https://httpwg.org/specs/rfc7230.html#rfc.section.3.1.2): _The reason-phrase element exists for the sole purpose of providing a textual description associated with the numeric status code, mostly out of deference to earlier Internet application protocols that were more frequently used with interactive text clients. A client SHOULD ignore the reason-phrase content._

I understand the fact that "OK" is counterintuitive for a 500 response. However we would rather not introduce complexity to our code by generating reason-phrases that match users' expectation that is actually untrue. IMO we should rather try to figure out if we can use the opportunity to educate users that they should not rely on the reason-phrase.

**Comment by tayler1** [comment]
Yes, this is good explanation, but only for world of machines. They don't rely on anything and just work.

While most of humans unable to correctly spot differences between 
"200 OK" and "500 OK"
"411 OK" and "417 OK"
Just imagine that some of responses passed from backend (and passed properly accidentally hahaha!):
"200 OK" "500 OK" "500 OK" "200 OK" "500 Internal Server Error"
What? This is a mess... literally.


I was wrong when initially describe the bug as Rack-related only, there are incorrect codes throughout all your code.
https://github.com/h2o/h2o/blob/2a44f945b9bb99865576ae10b9d85b2d78d8f4d7/lib/core/request.c#L154
https://github.com/h2o/h2o/blob/2a44f945b9bb99865576ae10b9d85b2d78d8f4d7/lib/core/request.c#L716

RFC 7230 says that description has no meaning and should be ignored (only by machines, not by humans!)
But RFC 7231 says the status code is "404 Not Found", this means all other text variations is perfectly incorrect.
